packages/visualizations/src/components/Chart/Chart.svelte

```diff
@@ -156,14 +161,22 @@
     let displaySubtitle: boolean;
     $: displayTitle = defaultValue(options?.title?.display, !!options?.title?.text);
     $: displaySubtitle = defaultValue(options?.subtitle?.display, !!options?.subtitle?.text);
+
+    let legendPosition: LegendPositions;
+    $: legendPosition =
+        clientWidth <= 375 ? 'bottom' : defaultValue(options?.legend?.position, 'bottom');
+    let legendOptions: CategoryLegendType;
+    $: if (options?.legend?.custom) {
+        legendOptions = buildCustomLegend({ chart, options, chartConfig });
+    }
 </script>
 
 {#if data.error}
     Error : {JSON.stringify(data.error)}
 {:else if options}
-    <figure>
+    <div bind:clientWidth class="container">
```

Comment:
I'm just wondering if is there a way to compute clientWidth and bind it conditionally as for now we only use it for custom legend but we process it for all charts ?